### DISCUSSION TITLE
Linking errors with backends and vcpkg

### DISCUSSION THREAD

**Discussion by Torino1564** [ocornut/imgui#7168] [New Users Build/Link/Run issues ONLY!]
Good evening. Im in the process of switching to using vcpkg as a packet manager instead of building the source code myself, and Im having trouble with the backends. I have my static library that a vcpkg.json property file that adds imgui as a dependency with its respective backends.
`"name": "tryn-core",
"version": "0.1.0",
"dependencies": [
  "assimp",
  "cli11",
  {
    "name": "imgui",
    "features": [
      "docking-experimental",
      "win32-binding",
      "dx11-binding",
      "dx12-binding"
    ],
    "platform": "(windows & x64)"
  }
]`
Im then linking this library to my executable project, and thats where Im getting the linker errors regarding the backend function calls. I dont have any issues with the imgui library itself nor with any other library, but the moment I tried to use vcpkg for the backends thats when I got this issue. Any suggestions as of what I could be doing wrong are greatly appreciated it.

**Comment by bikemurt** [comment]
This config worked for me.

vcpkg.json:

```
{
    "name": "vcpkg-test",
    "version": "0.0.0",
    "dependencies": [
        {
            "name": "imgui",
            "version>=": "1.90.7",
            "features": ["glfw-binding", "opengl3-binding"]
        },
        {
            "name": "glfw3",
            "version>=": "3.4"
        },
        {
            "name": "glew",
            "version>=": "2.2.0#3"
        }
    ],
    "builtin-baseline": "f7423ee180c4b7f40d43402c2feb3859161ef625"
}
```

Then include the following headers in your code:
```
#include "GL/glew.h"
#include "GLFW/glfw3.h"
#include "imgui.h"
#include "imgui_impl_glfw.h"
#include "imgui_impl_opengl3.h"
```

Edit: the backends aren't stored in the typical `backends/` folder that you get when you do it this way with FetchContent, just the root of your build's src folder

```

FetchContent_Declare(
        imgui
        GIT_REPOSITORY https://github.com/ocornut/imgui
        GIT_TAG v1.90.7
)
```
